src/parquet/column/reader.h

```diff
@@ -47,6 +47,66 @@ struct hash<parquet::Encoding::type> {
 
 namespace parquet_cpp {
 
+class LevelDecoder {
```

Comment:
Can you place this class in its own header file `src/parquet/column/levels.h`. We will also start a unit test suite for the level encoders and decoders in `src/parquet/column/levels-test.cc`